### PR TITLE
INC new API tensorflow examples update

### PR DESCRIPTION
**Pull request by lvliang-intel**
Signed-off-by: Lv, Liang1 <liang1.lv@intel.com>

## Type of Change

feature
API changed 

## Description

detail description 
JIRA ticket: ILITV-2456
INC New API TF examples(resnetv2_50,xception,inception_resnet_v2,resnet_v2)

## Expected Behavior & Potential Risk

Tensorflow examples with partial build-in func or without build-in func, update to new API and Config class.

## How has this PR been tested?

UT, Pre-CI test.

## Dependency Change?

No.

### CODE REVIEW

**Comment by chensuyue** [comment]
1. delete config yaml. 
2. unify data input with "--dataset_location=", we use the same dataset for calib and eval. 
3. benchmark require "--mode=(accuracy/performance)",  BenchmarkConfig is performance only. 
if mode==“accuracy”: 
	eval_func()
else:
	benchmark.fit(BenchmarkConfig(...), b_dataloader…)
4. Need to unify the why to handle ITEX gpu case. How to set gpu without yaml, with input param or ask user to add device=gpu in code?

File: examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py [hunk 1]
```diff
@@ -59,70 +52,96 @@
...
+            model=FLAGS.input_model,
+            conf=config,
+            calib_dataloader=calib_dataloader,
+            calib_func=None,
+            eval_dataloader=eval_dataloader,
+            eval_func=evaluate,
+            eval_metric=None)
+        q_model.save(FLAGS.output_model)
+
+    if FLAGS.benchmark:
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py:140]: please add benchmark for accuracy.

File: examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/run_benchmark.sh [hunk 2]
```diff
@@ -13,17 +13,14 @@ function init_params {
   for var in "$@"
   do
     case $var in
-      --config=*)
-          config=$(echo $var |cut -f2 -d=)
-      ;;
       --input_model=*)
           input_model=$(echo $var |cut -f2 -d=)
       ;;
       --mode=*)
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/run_benchmark.sh:19]: 'mode' will input accuracy or performance.

**Comment by chensuyue** [comment]
## extension test 
newAPI: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3596/artifact/report.html 
master oldAPI: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3598/artifact/report.html 
### issues
1. Tuning time consuming gap.   
2. Please support benchmark for accuracy.

File: examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/run_benchmark.sh [hunk 3]
```diff
@@ -13,17 +13,14 @@ function init_params {
...
           mode=$(echo $var |cut -f2 -d=)    
       ;;
-      --eval_data=*)
-          eval_data=$(echo $var |cut -f2 -d=)    
+      --dataset_location=*)
+          dataset_location=$(echo $var |cut -f2 -d=)    
       ;;
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/run_benchmark.sh:24]: image
Please set 4 parameters for run_benchmark.sh

File: examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py [hunk 4]
```diff
@@ -59,70 +52,76 @@
...
 
 eval_dataset = TensorflowImageRecord(root=FLAGS.eval_data, transform=ComposeTransform(transform_list= \
-  [BilinearImagenetTransform(height=299, width=299)]))
+                 [BilinearImagenetTransform(height=299, width=299)]))
 if FLAGS.benchmark and FLAGS.mode == 'performance':
-  eval_dataloader = DefaultDataLoader(dataset=eval_dataset, batch_size=1)
+    eval_dataloader = DefaultDataLoader(dataset=eval_dataset, batch_size=1)
 else:
-  eval_dataloader = DefaultDataLoader(dataset=eval_dataset, batch_size=32)
+    eval_dataloader = DefaultDataLoader(dataset=eval_dataset, batch_size=32)
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py:59]: please set benchmark bs with input batch_size.

File: examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py [hunk 5]
```diff
@@ -59,70 +52,76 @@
...
+    if FLAGS.benchmark:
+        from neural_compressor.benchmark import fit
+        from neural_compressor.config import BenchmarkConfig
+        if FLAGS.mode == 'performance':
+            conf = BenchmarkConfig(iteration=100, cores_per_instance=4, num_of_instance=7)
+            fit(FLAGS.input_model, conf, b_dataloader=eval_dataloader, b_func=evaluate)
+        else:
+            from neural_compressor.experimental import common
+            accuracy = evaluate(common.Model(FLAGS.input_model).model)
+            print("Accuracy: %.5f" % accuracy)
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/inception_resnet_v2/quantization/ptq/main.py:124]: Please print bs for accuracy as well.

File: neural_coder/__main__.py [hunk 6]
```diff
@@ -55,11 +55,19 @@ def parse_args():
 from neural_coder import enable
 if args.opt == "":
     if args.approach == "static":
+<<<<<<< HEAD
```
Comment (chensuyue) [inline neural_coder/__main__.py:58]: Pls handle the conflict.

**Comment by chensuyue** [comment]
## Gaps between oldAPI and newAPI.  
old: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3598/artifact/linux/tensorflow/resnetv2_50_keras/tensorflow-resnetv2_50_keras-linux-clx8280-tune.log  
new: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3605/artifact/linux/tensorflow/resnetv2_50_keras/tensorflow-resnetv2_50_keras-linux-clx8280-tune.log   
1. newAPI tuning time is much longer than oldAPI. 
image
image
2. config is the same, but int8 accuracy is different. 
3. 'Dump user yaml configuration' seems not correct for newAPI.

File: examples/tensorflow/image_recognition/keras_models/resnetv2_50/quantization/ptq/main.py [hunk 7]
```diff
@@ -59,70 +54,77 @@
...
+            calib_dataloader=calib_dataloader,
+            eval_dataloader=eval_dataloader,
+            eval_func=evaluate)
+        q_model.save(FLAGS.output_model)
+
+    if FLAGS.benchmark:
+        from neural_compressor.benchmark import fit
+        from neural_compressor.config import BenchmarkConfig
+        if FLAGS.mode == 'performance':
+            conf = BenchmarkConfig(iteration=100, cores_per_instance=4, num_of_instance=7)
```
Comment (chensuyue) [inline examples/tensorflow/image_recognition/keras_models/resnetv2_50/quantization/ptq/main.py:121]: iteration doesn't work, https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3605/artifact/linux/tensorflow/resnetv2_50_keras/tensorflow-resnetv2_50_keras-fp32-throughput-linux-clx8280.log

**Review by zehao-intel — APPROVED** [review]

**Comment by chensuyue** [comment]
Extension test: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3630/artifact/report.html

**Comment by chensuyue** [comment]
[extension test](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3705/artifact/report.html) 
pls check the performance gap.

**Comment by chensuyue** [comment]
[extension test](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3717/artifact/report.html) 
minor issue need to fix, and pls rebuild the extension job with the failed model after bug fix.

**Review by hshen14 — APPROVED** [review]
LGTM

**Comment by chensuyue** [comment]
still bas some [performance gap](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3727/artifact/report.html), need to double check.

**Comment by chensuyue** [comment]
[extension test](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3772/)

**Review by chensuyue — APPROVED** [review]
